cloudslang-compiler/src/main/java/io/cloudslang/lang/compiler/validator/PreCompileValidatorImpl.java

```diff
@@ -308,9 +300,17 @@ private String getMessagePart(Class aClass) {
         return messagePart;
     }
 
-    private String getExecutableName(Map<String, Object> executableRawData) {
+    private String getExecutableName(Map<String, Object> executableRawData, List<RuntimeException> errors) {
         String execName = (String) executableRawData.get(SlangTextualKeys.EXECUTABLE_NAME_KEY);
-        return execName == null ? "" : execName;
+        if (StringUtils.isBlank(execName)) {
+            errors.add(new RuntimeException("Executable has no name"));
+        }
+        try {
+            executableValidator.validateExecutableName(execName);
```

Comment:
I think it's better this way, since ExecutableValidator has the specific validate methods, whereas AbstractValidator has general logic, in the future we may want to aplpy other validation rules to exec name than `simpleNameRules`